src/mono/mono/mini/aot-compiler.c

```diff
@@ -14284,7 +14284,7 @@ acfg_free (MonoAotCompile *acfg)
 	g_free (acfg->static_linking_symbol);
 	g_free (acfg->got_symbol);
 	g_free (acfg->plt_symbol);
-	g_free (acfg->global_prefix);
+	g_free (acfg->assembly_name_sym);
```

Comment:
we need to free both global_prefix and assembly_name_sym (not freeing assembly_name_sym is an existing issue)